rust/datafusion/src/execution/physical_plan/hash_aggregate.rs

```diff
@@ -327,120 +278,47 @@ impl RecordBatchReader for GroupedHashAggregateIterator {
                 })
                 .collect::<ArrowResult<Vec<_>>>()?;
 
-            // create vector large enough to hold the grouping key
+            // create vector to hold the grouping key
             let mut key = Vec::with_capacity(group_values.len());
             for _ in 0..group_values.len() {
-                key.push(GroupByScalar::UInt32(0));
+                key.push(KeyScalar::UInt32(0));
             }
 
             // iterate over each row in the batch and create the accumulators for each grouping key
-            let mut accumulators: Vec<Rc<AccumulatorSet>> =
```

Comment:
It isn't clear, but this is actually an optimization. By buiding the vec of accumulators, it allows better columnar processing and presumably is making better use of the cpu cache. This PR shows the same minor slow down in performance as the other branch I created.

Before:

```
Running benchmarks with the following options: Opt { debug: false, iterations: 3, batch_size: 4096, path: "/mnt/nyctaxi/csv/year=2019", file_format: "csv" }
Executing 'fare_amt_by_passenger'
Query 'fare_amt_by_passenger' iteration 0 took 11058 ms
Query 'fare_amt_by_passenger' iteration 1 took 11314 ms
Query 'fare_amt_by_passenger' iteration 2 took 11479 ms
```

After

```
Running benchmarks with the following options: Opt { debug: false, iterations: 3, batch_size: 4096, path: "/mnt/nyctaxi/csv/year=2019", file_format: "csv" }
Executing 'fare_amt_by_passenger'
Query 'fare_amt_by_passenger' iteration 0 took 11636 ms
Query 'fare_amt_by_passenger' iteration 1 took 12500 ms
Query 'fare_amt_by_passenger' iteration 2 took 12358 ms
```

This is from running the benchmark crate in this repo.
```

Reply:
Mind blowing. Still much to learn.

I struggle to run the benchmarks on my computer. Is there any design reason to not run the benchmarks as part of the pipeline? Are they too unstable against changes in hardware?

Reply:
Ok, I was able to run the benchmarks, but I do not obverse any statistically significant difference:

```
# the commit pre on master:
git checkout cd503c3f583dab4b94c9934d525664e5897ff06d 
cargo bench
```

leaves me with

```
aggregate_query_no_group_by                                                                            
                        time:   [121.39 us 121.55 us 121.73 us]
Found 9 outliers among 100 measurements (9.00%)
  3 (3.00%) high mild
  6 (6.00%) high severe

aggregate_query_group_by                                                                            
                        time:   [170.22 us 170.75 us 171.47 us]
Found 12 outliers among 100 measurements (12.00%)
  7 (7.00%) high mild
  5 (5.00%) high severe

aggregate_query_group_by_with_filter                                                                            
                        time:   [279.00 us 279.34 us 279.71 us]
Found 9 outliers among 100 measurements (9.00%)
  2 (2.00%) high mild
  7 (7.00%) high severe
```

followed by 

```
# the latest commit on this branch:
git checkout bbd9da7ce5b582587bce5c8ff8a228f5425e0113
cargo bench
```

leaves me with

```
aggregate_query_no_group_by                                                                            
                        time:   [122.19 us 122.38 us 122.58 us]
                        change: [-1.5876% +0.3639% +2.3348%] (p = 0.74 > 0.05)
                        No change in performance detected.
Found 6 outliers among 100 measurements (6.00%)
  6 (6.00%) high severe

aggregate_query_group_by                                                                            
                        time:   [172.66 us 172.91 us 173.19 us]
                        change: [-0.9329% +1.2144% +3.2272%] (p = 0.28 > 0.05)
                        No change in performance detected.
Found 12 outliers among 100 measurements (12.00%)
  6 (6.00%) high mild
  6 (6.00%) high severe

aggregate_query_group_by_with_filter                                                                            
                        time:   [282.30 us 282.70 us 283.14 us]
                        change: [-1.1013% +0.9700% +2.7683%] (p = 0.40 > 0.05)
                        No change in performance detected.
Found 8 outliers among 100 measurements (8.00%)
  2 (2.00%) high mild
  6 (6.00%) high severe
```

Reply:
@jorgecarleitao Out of curiousity what are the hardware specs you ran this on?

Reply:
Mac OS 10.14.5, 2.3 GHz Intel Core i5, 8 GB 2133 MHz LPDDR3.

Reply:
I would be interested in finding out what AVX instruction sets are supported on the specific CPU that you have. For example, does it support AVX2?

Reply:
I do not think so:

```
$ sysctl -a | grep machdep.cpu.features
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX SMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
```

There is a `AVX1.0` there.

Reply:
Actually, this information is incorrect. My CPU family is [Kaby Lake](https://en.wikipedia.org/wiki/Kaby_Lake), which as the AVX2.0 extension.